tutorials/rag/evaluate-rag-flow.ipynb

```diff
@@ -842,6 +832,277 @@
     "\n",
     "![Eval Logs table](../../assets/images/evaluate_rag_flow_logs.png)"
    ]
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Using Retriever as a Tool\n",
+    "\n",
+    "In the above example, the flow of our app is static - given the question we always call the retriever, pass the context to the model and then respond to the user. It can be helpful to have a more dynamic setup where we instead allow the model to choose when use the retriever.\n",
+    "This allows the model to do smart query expansion, or ask for clarifying questions, before deciding to retrieve the context which can be significant boost in performance.\n",
```

Comment:
this sentence is awkward. i think it's missing a comma before "which". it's at least missing a "a" before "significant".

Reply:
Oh this should not be here anymore, pulled it into a script